lib/acpi/acpi.c

```diff
@@ -461,7 +461,6 @@ int acpi_device_irq_get(struct acpi_dev *child_dev, struct acpi_irq_resource *ir
 			return -ENOMEM;
 		}
 
-		memset(irq_res, 0, sizeof(struct acpi_irq_resource));
 		irq_res->irq_vector_max = res->Data.ExtendedIrq.InterruptCount;
```

Comment:
What do you do if `res->Data.ExtendedIrq.InterruptCount` is greater than `CONFIG_ACPI_IRQ_VECTOR_MAX`? Don't you get a buffer overrun in that case?

I think the simple solution (which wouldn't require API changes) would be for the caller to pass the array size through the initial values of `irq_vector_max` and `mmio_max` and then upon return these would contain the actual number of fetched elements.

Reply:
updated